### PR TITLE
fix(popup): Add minimum height to popup

### LINKED ISSUE #1534: Toolbar window: Minimum height has gone missing
### 🐛 Describe the bug

The toolbar window now looks like this with no recent entries:

Screenshot 2019-10-28 at 11 04 17

### Expected behaviour

It should have some kind of height, it used to be `300px` or something like that, I think. A change to the way browser extensions work has probably broken it.

I believe this is usually set in the Manifest, but I can't quite remember. Maybe it was a minimum height in the content.

### Steps to reproduce

1. Install in Chrome
2. Log into account with no recent entries or few recent entries
3. See that the window is not very tall when you open it.

### PR DESCRIPTION
**Pull request by lise-toggl**
## :star2: What does this PR do?

Specifies a minimum height for the Toggl Button popup so that it doesn't look all squashed for accounts with no or few entries:
image

## :bug: Recommendations for testing

1. Login to the Button with an account with 0 entries

## :memo: Links to relevant issues or information

Closes #1534.

WIP: waiting for clarification on whether the "See more on ..." footer should always be displayed or not (or some other placeholder message for accounts without entries).

### CODE REVIEW

**Comment by shantanuraj** [comment]
I think we always want to display the see more, it opens the webapp and can be used to do stuff not possible with the extension, eg. reports

**Comment by lise-toggl** [comment]
Done, now always displays the footer message:

image

**Comment by shantanuraj** [comment]
Let's push the See more button to the bottom
and to make some use of the middle section  let's reuse webapp's empty state onboarding image and copy

## webapp
image

#### Image
toggl-stopwatch

#### Copy
"Get ready to track time and boost your productivity!"

File: src/scripts/components/TimeEntriesList.tsx [hunk 1]
```diff
@@ -35,11 +35,23 @@ const getTimeEntryDayGroups = (timeEntries: Array<Array<Toggl.TimeEntry>>): {[da
...
+    </a>
+  </Footer>);
+
 interface TimeEntriesListProps {
   timeEntries: Toggl.TimeEntry[][];
   projects: IdMap<Toggl.Project>;
 };
-export default function TimeEntriesList (props: TimeEntriesListProps) {
+export default function TimeEntriesList (props?: TimeEntriesListProps) {
+  if(Object.keys(props).length === 0) {
```
Comment (shantanuraj) [inline src/scripts/components/TimeEntriesList.tsx]: TS is throwing an error here

Comment (shantanuraj) [reply src/scripts/components/TimeEntriesList.tsx]: PS. I think you don't really need this

File: src/scripts/components/TimeEntriesList.tsx [hunk 2]
```diff
@@ -35,11 +35,23 @@ const getTimeEntryDayGroups = (timeEntries: Array<Array<Toggl.TimeEntry>>): {[da
...
+      </Button>
+    </a>
+  </Footer>);
+
 interface TimeEntriesListProps {
   timeEntries: Toggl.TimeEntry[][];
   projects: IdMap<Toggl.Project>;
 };
-export default function TimeEntriesList (props: TimeEntriesListProps) {
+export default function TimeEntriesList (props?: TimeEntriesListProps) {
```
Comment (shantanuraj) [inline src/scripts/components/TimeEntriesList.tsx]: props will never be `undefined`, so if you want to mark things as optional mark the keys in L48 and L49

File: src/scripts/components/TimeEntriesList.tsx (hunk 1 above)
Comment (shantanuraj) [reply src/scripts/components/TimeEntriesList.tsx]: Line 55 below deals with this, it sets `timeEntries` to an empty array, so you move the if statement after it
and change it something like

```typescript
if (timeEntries.length === 0) {
  return <TimeEntriesFooter />;
}
```

Comment (shantanuraj) [reply src/scripts/components/TimeEntriesList.tsx]: Though in this case you might want something like

```typescript
if (timeEntries.length === 0) {
  return (
    <React.Fragment>
      <EmptyState />
      <TimeEntriesFooter />
    </React.Fragment>
  );
}
```

**Comment by lise-toggl** [comment]
How's this?

image

image

**Comment by shantanuraj** [comment]
Perfect 👌

File: src/scripts/components/TimeEntriesList.tsx (hunk 1 above)
Comment (lise-toggl) [reply src/scripts/components/TimeEntriesList.tsx]: updated :raised_hands:

File: src/scripts/components/TimeEntriesList.tsx (hunk 2 above)
Comment (lise-toggl) [reply src/scripts/components/TimeEntriesList.tsx]: reverted :raised_hands:

**Comment by lise-toggl** [comment]
Or maybe we could move down the message a bit? :eyes:

**Comment by shantanuraj** [comment]
That works as well 👍

**Comment by lise-toggl** [comment]
Done ＼(★^∀^★)／

image

**Pull request merged**